west_commands: pyocd: issue a warning when falling back to binary file

SIGSTACKFAULT opened the pull request:
Signed-off-by: Connor King <SIGSTACKFAULT@gmail.com>

mbolivar commented on scripts/west_commands/runners/pyocd.py:
```diff
@@ -111,15 +111,20 @@ def do_run(self, command, **kwargs):
+            return self.hex_name
+
+        if os.path.isfile(self.bin_name):
+            self.logger.warning('Hex file "{}" not found, falling back to binary "{}"'.format(self.hex_name, self.bin_name))
```
Please keep this file 80-column clean.

mbolivar replied:
Why was this comment marked resolved? Did you forget to push an update?

SIGSTACKFAULT replied:
I have no idea what I'm doing. Never used this feature.

SIGSTACKFAULT replied:
Thought resolving would apply the suggestion

mbolivar commented on scripts/west_commands/runners/pyocd.py:
```diff
@@ -132,7 +137,7 @@ def flash(self, **kwargs):
                [fname])
 
-        self.logger.info('Flashing Target Device')
+        self.logger.info('Flashing Target Device ({})'.format(os.path.basename(fname)))
```
Here too.

mbolivar reviewed (changes requested):
Looks good, just some whitespace nits. Fine for 2.0 in my opinion with those fixed.

SIGSTACKFAULT commented:
> 1: UC2 Signed-off-by: must have a full name

yes it does!

mbolivar commented:
>>    1: UC2 Signed-off-by: must have a full name
> yes it does!

My guess is the "SIGSTACKFAULT" is causing the issue; try removing that maybe?

SIGSTACKFAULT committed:
west_commands: pyocd: issue a warning when falling back to binary file

Some tutorial told me to do `west flash --hex-file zephyr.signed.hex`;
  west silently falls back to `build/zephyr/zephyr.bin`.

TL;DR:
  If the provided hex file doesn't exist, issue a warning.
  (or if the cache is bad, since there's no way to tell)

Signed-off-by: Connor King <SIGSTACKFAULT@gmail.com>

SIGSTACKFAULT commented:
gotta wait for all the checks to pass before I can read gitlint's output, grumble grumble.

SIGSTACKFAULT commented:
@mbolivar I've fixed the line length stuff.

mbolivar commented:
Looks like you need a rebase now that https://github.com/zephyrproject-rtos/zephyr/pull/18310 is merged

mbolivar commented:
> gotta wait for all the checks to pass before I can read gitlint's output, grumble grumble.

You can run it locally. See the Contribution Guidelines in the documentation.

mbolivar commented:
@sigstackfault ping

SIGSTACKFAULT committed:
Merge branch 'master' into pyocd_runner_fix

SIGSTACKFAULT commented:
Welp, I guess this is just redundant.
I should check my email more often.
